compute_endpoint/globus_compute_endpoint/endpoint/endpoint.py

```diff
@@ -371,6 +371,16 @@ def start_endpoint(
             log.critical(msg)
             raise ValueError(msg)
 
+        if endpoint_config.high_assurance and not getattr(
+            endpoint_config.engine, "encrypted", False
+        ):
+            msg = (
```

Comment:
Perhaps only do this check if the engine has a notion of encryption?  Consider:

```python
if endpoint_config.high_assurance and hasattr(endpoint_config.engine, "encrypted"):
    if not endpoint_config.engine.encrypted:
        ...
```
Let's see if mypy likes the inner if condition ...

Would want to add a test against the other known engines as well.

Reply:
Following some side-channel back and forth, decided to go with a method that each engine must implement to indicate whether or not it's HA compliant.